### ISSUE TITLE
format tool does not ignore ./libs

### ISSUE THREAD

**Issue by bcardiff** [crystal-lang/crystal#3267]
When using a project with dependencies, `crystal deps` will end up installing files in `./libs`. 
This prevents using `$ crystal tool format --check` to check the whole source code as part of the ci in projects with dependencies.

proposals:
- a) `./libs`should be excluded. (fixed)
- b) add `--exclude PATTERN` with default to exclude ./libs
- c) add `--exclude PATTERN` with no default

I reach this while trying to [add the format --check to crystal-mysql](https://travis-ci.org/crystal-lang/crystal-mysql/builds/157924744)

**Comment by kostya** [comment]
may be create config file `.crystal` in project home when crystal generate it. and add option by default
```yaml
formatter:
  exclude:
    - ./libs
```

**Comment by asterite** [comment]
Right, some of the options should be considered.

As a workaround you can try to see if `crystal tool format src --check` works in this case.

**Comment by bcardiff** [comment]
Yes, I can split the check per directory.

```
crystal tool format src --check
crystal tool format spec --check
```

(or force formatting in dependable project, as long as it can be applied)
But it's just a use case that is not polished: formatter + deps.

**Comment by asterite** [comment]
You can also pass multiple directories :-)

```cr
crystal tool format src spec --check
```

But I agree that maybe the default should skip `libs`.

**Comment by sdogruyol** [comment]
I also vote for skippling `libs`

**Comment by ysbaddaden** [comment]
We can already specify a list of folders and files; the include argument feels redundant. Overall this looks complex for no real benefit 

Why not just specify the folders to format? instead of specifying additional? What about defaulting to the standard locations (i.e. "src spec") instead of "." when no folders are specified?

**Comment by straight-shoota** [comment]
I had this idea in [#4635 (comment)[4635#issuecomment-315551930] as well (last item). For most basic projects this should be totally fine and would be easy to implement.

But with more complicated project setups this might not be sufficient. I'm thinking about formatting sample code in `samples/` but ignoring `samples/*/lib`. I don't know if this is such a big use-case (at least I have it this way in [crinja](https://github.com/straight-shoota/crinja)) and it might not be necessary to provide this feature directly but I feel like this should be possible somehow. An external solution would be to allow `crystal tool format` to receive a list of files from stdin. Then you could use `find` or other tools (like `git ls-files`) to provide a selection of to format.

**Comment by straight-shoota** [comment]
This has been fixed by #4635
